src/instance.rs

```diff
@@ -119,15 +118,17 @@ impl<T> Py<T> {
     }
 }
 
-pub trait AsPyRef<T: PyTypeInfo>: Sized {
+pub trait AsPyRef: Sized {
+    type Target;
     /// Return reference to object.
-    fn as_ref(&self, py: Python) -> &T;
+    fn as_ref(&self, py: Python<'_>) -> &Self::Target;
 }
 
-impl<T: PyTypeInfo> AsPyRef<T> for Py<T> {
-    fn as_ref(&self, _py: Python) -> &T {
+impl<'p, T: PyClass> AsPyRef for Py<T> {
```

Comment:
Maybe also need `impl<'p, T: AsPyPointer> AsPyRef for Py<T>` ?

Reply:
This would cause `conflicting implementation` error.
I placed implementations for native types in `types/mod.rs`.

Reply:
Ah yes, of course. Then how about 

```
impl<'p, T: PyTypeInfo> AsPyRef for Py<T> {
    type Target = <T as PyTypeInfo>::Layout;
    ...
}
```

Reply:
Layout is `ffi::PyObject` or so for native types.
It's not convenient to expose them.

Reply:
Ah - I thought their layout was `PyAny`, my bad. Ok let's stick with the current design :)